src/github.com/getlantern/flashlight/client/reverseproxy.go

```diff
@@ -40,23 +28,25 @@ func (client *Client) initReverseProxy(bal *balancer.Balancer, dumpHeaders bool)
 		DisableKeepAlives: true,
 	}
 
+	// Just choose a random dialer that also takes care of things like the
+	// authentication token.
+	dialer := client.getBalancer().RandomDialer()
```

Comment:
There are other things that `balancer.Dial` did, like sending plaintext traffic only to trusted servers and retrying with another dialer in case the first random pick did not work, with `RandomDialer()` we end up with a dialer that may or may not work at all, I imagine this could case trouble when doing lots of requests, like querying small files to display a web page; as I understand each web page resource will require a new random connection.